weave/scorers/context_relevance_scorer.py

```diff
@@ -0,0 +1,382 @@
+   - Is unnecessary content avoided?
+
+2. **Structural Coherence**
+   - Does the output flow logically and show internal consistency?
+   - Is the presentation of information clear and organized?
+   - Is there a good balance between completeness and conciseness?
+
+3. **Contextual Integration**
+   - How well does the output use the provided context?
+   - Does the output align with the broader discourse?
+   - Is it consistent with background information?
+   - Does it fulfill task-specific requirements?
+
+## Evaluation Process
+
+1. Review all input context (instructions, prompts, documents, chat history)
+2. Identify core requirements and purpose
+3. Analyze the LLM output across all three dimensions
+4. Assign a single relevance score (1-5):
+   - 5: Exceptional relevance across all dimensions
+   - 4: Strong relevance with minor gaps
+   - 3: Adequate relevance with some issues
+   - 2: Significant relevance issues
+   - 1: Major relevance problems
+5. Classify as relevant (score ≥ 3.5) or not relevant (score < 3.5)
+
+## Task-Specific Considerations
+
+- **Summarization**: Focus on key information selection and density
+- **Q&A**: Emphasize answer accuracy and completeness
+- **Chat**: Consider conversation flow and context maintenance
+- **RAG**: Evaluate retrieved information integration
+
+## Output Format
+
+Provide evaluation results in the following JSON format:
+
+```json
+{
+  "relevance": [score from 1-5],
+  "relevant": [true/false]
+}
+```
+"""
+
+
+class OldRelevanceScorer(Scorer):
+    """
+    Use wandb/relevance_scorer to check if the model output is relevant.
+
+    Args:
+        model_name: The name of the relevance scorer model to use. Defaults to `wandb/relevance_scorer`.
+        device: The device to use for inference. Defaults to `None`, which will use `cuda` if available.
+    """
+
+    model_name_or_path: str = None
+    base_url: Optional[str] = None
+    device: str = None
+    _classifier: Any = PrivateAttr()
+    _tokenizer: Any = PrivateAttr()
+    _id2label: dict[int, str] = PrivateAttr()
+    _system_prompt: str = PrivateAttr()
+
+    def model_post_init(self, __context: Any) -> None:
+        try:
+            import torch
+            from transformers import pipeline
+        except ImportError:
+            print(
+                "The `transformers` package is required to use the ContextRelevanceScorer, please run `pip install transformers`"
+            )
+        if self.base_url:
+            print(f"Using external API at {self.base_url} for scoring.")
+            return  # Skip local model loading if base_url is provided
+
+        """Initialize the coherence model and tokenizer."""
+        self.device = set_device(self.device)
+        if os.path.isdir(self.model_name_or_path):
+            self._local_model_path = self.model_name_or_path
+        else:
+            self._local_model_path = download_model(
+                scorer_model_paths["relevance_scorer"]
+            )
```

Comment:
I see the same code block in a few model based scorers. We can move this to a function `get_local_model_path`.

Additionally, this line:

`if os.path.isdir(self.model_name_or_path):` fails because `model_name_or_path` is None as default.